File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.swift
class Solution {
    func minSubArrayLen(_ target: Int, _ nums: [Int]) -> Int {
        var minLength = Int.max
        var windowStart = 0 
        var sum = 0
        for windowEnd in 0..<nums.count{
            sum += nums[windowEnd]
            
            while(sum >= target){
                minLength = min(minLength, windowEnd - windowStart + 1)
                sum -= nums[windowStart]
                windowStart += 1
            }
        }
        if minLength == Int.max{
            return 0
        }
        return minLength
    }
}

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int ans = INT_MAX, windowStart = 0, sz = nums.size(), rangeSum = 0;
        
        for(int windowEnd = 0; windowEnd < sz; windowEnd ++ ){
            rangeSum += nums[windowEnd];
            
            while(rangeSum >= target){
                ans = min(ans, windowEnd - windowStart + 1);
                rangeSum -= nums[windowStart];
                windowStart ++ ;
            }
        }
        
        return (ans == INT_MAX) ? 0 : ans;
        
    }
};